src/Bicep.Core/Emit/TemplateWriter.cs

```diff
@@ -109,11 +109,16 @@ public void Write(SourceAwareJsonTextWriter writer)
 
             emitter.EmitProperty("$schema", GetSchema(Context.SemanticModel.TargetScope));
 
-            if (Context.Settings.EnableSymbolicNames)
+            if (Context.Settings.EnableSymbolicNames && !Context.Settings.EnableAsserts)
             {
                 emitter.EmitProperty("languageVersion", "1.10-experimental");
             }
 
```

Comment:
The reasoning behind this requires context which future people working on this code may not have - please add a comment to explain the reason for the language version differentiation.

Reply:
I just added a comment above this code